<?xml version="1.0"?>
<launch>
	<arg name="device" default="/dev/video0" doc="Device path"/>
	<arg name="width" default="640" doc="Image width in pixels"/>
	<arg name="height" default="480" doc="Image height in pixels"/>
	<arg name="config_file" default="$(find v4l2_camera)/config/calibration.yaml"/>

	<node name="v4l2_camera" pkg="v4l2_camera" type="v4l2_camera" output="screen" args="$(arg device)">

		<param name="camera_info" type="string" value="$(arg config_file)" />
		<param name="camera_frame" value="v4l2_camera" type="string" />
		<rosparam command="load" file="$(arg config_file)" />
	</node>

	<node name="v4l2_camera_view" pkg="image_view" type="image_view" output="screen">
  		<remap from="image" to="v4l2_camera/image_raw"/>
	</node>
</launch>

